code/modules/unit_tests/modify_fantasy_variable.dm

```diff
@@ -1,21 +1,45 @@
 // Unit test to make sure that there are no duplicate keys when modify_fantasy_variable is called when applying fantasy bonuses.
 // Also to make sure the fantasy_modifications list is null when fantasy bonuses are removed.
+/datum/unit_test/modify_fantasy_variable
+
 /datum/unit_test/modify_fantasy_variable/Run()
+	var/list/applicable_types = subtypesof(/obj/item) - uncreatables
 
-	for(var/obj/item/path as anything in subtypesof(/obj/item))
+	for(var/obj/item/path as anything in applicable_types)
 		var/obj/item/object = allocate(path)
+		// objects will have fantasy bonuses inherent to their type (like butterdogs and the slippery component), so we need to take this into account
+		var/number_of_extant_bonuses = LAZYLEN(object.fantasy_modifications)
+
+		#define TEST_SUCCESS LAZYLEN(object.fantasy_modifications) == number_of_extant_bonuses
```

Comment:
```suggestion
#define TEST_SUCCESS LAZYLEN(object.fantasy_modifications) == number_of_extant_bonuses
```

Best practice in languages with C preprocessors is to not indent defines because it incorrectly suggests they have any sort of scope